File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int n = Integer.parseInt(br.readLine());

        Stack<Integer> stack = new Stack<>();
        StringTokenizer st;
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            String order = st.nextToken();
            int x = 0;
            if (order.equals("push")) {
                x = Integer.parseInt(st.nextToken());
                stack.push(x);
            } else if (order.equals("pop")) {
                if (!stack.empty()) {
                    sb.append(stack.pop()).append('\n');
                } else {
                    sb.append(-1).append('\n');
                }
            } else if (order.equals("size")) {
                sb.append(stack.size()).append('\n');
            } else if (order.equals("empty")) {
                if (!stack.empty()) {
                    sb.append(0).append('\n');
                } else {
                    sb.append(1).append('\n');
                }
            } else if (order.equals("top")) {
                if (!stack.empty()) {
                    sb.append(stack.peek()).append('\n');
                } else {
                    sb.append(-1).append('\n');
                }
            }
        }

        System.out.println(sb);
    }
}

File: 백준/Silver/10828. 스택/스택.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val sb = StringBuilder()
    val n = br.readLine().toInt()
    val stack = Stack<Int>()

    repeat(n) {
        val st = StringTokenizer(br.readLine())
        val op = st.nextToken()
        val x: Int

        when (op) {
            "push" -> {
                x = st.nextToken().toInt()
                stack.push(x)
            }

            "pop" -> {
                if (!stack.empty()) {
                    sb.append(stack.pop()).append('\n')
                } else {
                    sb.append(-1).append('\n')
                }
            }

            "size" -> {
                sb.append(stack.size).append('\n')
            }

            "empty" -> {
                if (!stack.empty()) {
                    sb.append(0).append('\n')
                } else {
                    sb.append(1).append('\n')
                }
            }

            "top" -> {
                if (!stack.empty()) {
                    sb.append(stack.peek()).append('\n')
                } else {
                    sb.append(-1).append('\n')
                }
            }
        }
    }

    print(sb)
}
